lab_1/main.py

```diff
@@ -70,6 +124,30 @@ def get_concordance(tokens: list, word: str, left_context_size: int, right_conte
     --> [['man', 'is', 'happy', 'the', 'dog', 'is'], ['dog', 'is', 'happy', 'but', 'the', 'cat']]
     """
     pass
+    if isinstance(right_context_size, bool) or isinstance(left_context_size, bool):     #check for emptiness
+        return []
+
+    if not isinstance(tokens, list) and not isinstance(word, str) \
+            and not isinstance(right_context_size, int) and not isinstance(left_context_size, int):       #check type
+        return []
+
+    part_text = []
+    for ind,elem in enumerate(tokens):  #make list with numbers & tokens
+        if elem == word:
+            if left_context_size > 0 and right_context_size > 0:
+                for i in ind:
+                    part_text.append(tokens[i - left_context_size:i+ right_context_size + 1])
+            elif left_context_size <= 0:
+                for i in ind:
+                    part_text.append(tokens[i:i + int(right_context_size) + 1])
+            elif right_context_size <= 0:
+                for i in ind:
+                    part_text.append(tokens[i - left_context_size:i + 1])
+            else:
+                part_text = []
+    return part_text
```

Comment:
left and right context logic can be better implemented and separated from main logic, but acceptable.